### PR TITLE
[16.0][FIX] stock_picking_auto_create_lot: immediate transfer with auto_create_lot option

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
Related issue: https://github.com/OCA/stock-logistics-workflow/issues/1281
Related odoo commit: https://github.com/odoo/odoo/commit/dd44e1f69dfd217fafa9c2ec5da8b211b2ff3513

@qrtl

### CODE REVIEW

File: stock_picking_auto_create_lot/models/stock_move.py [hunk 1]
```diff
@@ -22,3 +22,13 @@ def _compute_display_assign_serial(self):
...
+
+    # pylint: disable=missing-return
+    def _set_quantities_to_reservation(self):
+        super()._set_quantities_to_reservation()
+        for move in self:
+            if move.state not in ("partially_available", "assigned"):
+                continue
+            for move_line in move.move_line_ids:
+                if move.picking_type_id.auto_create_lot and move_line.lot_id:
+                    move_line.qty_done = move_line.reserved_uom_qty
```
Comment (yostashiro) [inline stock_picking_auto_create_lot/models/stock_move.py]:
```suggestion
            if move.product_id.tracking == "none" or not move.product_id.auto_create_lot or not move.picking_type_id.auto_create_lot:
                continue
            for move_line in move.move_line_ids:
                if move_line.lot_id:
                    # Create-backorder wizard would open without this.
                    move_line.qty_done = move_line.reserved_uom_qty
```

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: stock_picking_auto_create_lot/tests/test_stock_picking_auto_create_lot.py [hunk 2]
```diff
@@ -144,3 +144,22 @@ def test_multi_auto_create_lot(self):
...
+
+    def test_immediate_validate_tracked_move_with_auto_create_lot(self):
+        picking_form = Form(self.env["stock.picking"])
+        picking_form.picking_type_id = self.picking_type_in
+        with picking_form.move_ids_without_package.new() as move:
+            move.product_id = self.product_serial
+            move.product_uom_qty = 4
+        receipt = picking_form.save()
+        receipt.action_confirm()
+
```
Comment (yostashiro) [inline stock_picking_auto_create_lot/tests/test_stock_picking_auto_create_lot.py]: I'd consider to do something like the following, upon adjusting `_create_move()` to have the picking as an arg.
```suggestion
        picking_1 = self.picking
        self._create_move(product=self.product_serial, qty=4.0, picking=picking_1)
        picking_1.action_assign()
```

File: stock_picking_auto_create_lot/tests/test_stock_picking_auto_create_lot.py [hunk 3]
```diff
@@ -144,3 +144,22 @@ def test_multi_auto_create_lot(self):
...
+        immediate_wizard = receipt.button_validate()
+        immediate_wizard_form = Form(
+            self.env[immediate_wizard["res_model"]].with_context(
+                **immediate_wizard["context"]
+            )
+        ).save()
+        immediate_wizard_form.process()
+        # This test will fail before fix PR
```
Comment (yostashiro) [inline stock_picking_auto_create_lot/tests/test_stock_picking_auto_create_lot.py]:
```suggestion
        # Confirm that validation is not blocked, for example, by create-backorder wizard.
```

File: stock_picking_auto_create_lot/tests/test_stock_picking_auto_create_lot.py [hunk 4]
```diff
@@ -146,20 +145,19 @@ def test_multi_auto_create_lot(self):
...
-        with picking_form.move_ids_without_package.new() as move:
-            move.product_id = self.product_serial
-            move.product_uom_qty = 4
-        receipt = picking_form.save()
-        receipt.action_confirm()
-
-        immediate_wizard = receipt.button_validate()
+        # Clear existing move if not the picking will open backorder wizard because
+        # when we manually assign lot for serial_not_auto product, other products still
+        # have 0 done qty.
```
Comment (yostashiro) [inline stock_picking_auto_create_lot/tests/test_stock_picking_auto_create_lot.py]: I think we do not need to be so verbose as the intention should be clear.
```suggestion
        # Clear existing moves.
```

**Review by yostashiro — APPROVED** [review]
LGTM 👍

**Review by kanda999 — APPROVED** [review]
Functional Review : It worked as expected

**Pull request closed without merging**